python/lsst/pipe/tasks/findSatellites.py

```diff
@@ -0,0 +1,672 @@
+            if not (maskData == maskData.astype(bool)).all():
+                raise ValueError("binaryImage must be composed of 0s and 1s or booleans")
+
+        self.edges = self.cannyFilter(maskData)
+        self.lines = self.runKHT(self.edges)
+
+        if (binaryImage is None) and (self.config.imageBinning is not None):
+            # Renormalize line coordinates to original image:
+            self.lines['rho'] *= self.config.imageBinning
+            tmpMaskData = maskData.repeat(self.config.imageBinning, axis=0)
+            maskData = tmpMaskData.repeat(self.config.imageBinning, axis=1)
+        if len(self.lines) == 0:
+            lineMask = np.zeros_like(maskData, dtype=bool)
+            fitLines = np.empty(0, dtype=[('rho', float), ('theta', float), ('votes', int)])
+            clusters = np.empty(0, dtype=[('rho', float), ('theta', float), ('votes', int)])
+        else:
+            clusters = self.findClusters(self.lines)
+            fitLines, lineMask = self.fitProfile(clusters, maskedImage)
+
+        # The output mask is the intersection of the fit satellite trails and the image detections
+        outputMask = lineMask & maskData.astype(bool)
+
+        return pipeBase.Struct(
+            lines=fitLines,
+            lineClusters=clusters,
+            originalLines=self.lines,
+            mask=outputMask,
+        )
+
+    def processMaskedImage(self, maskedImage, checkMethod=False):
+        """Make binary image array from maskedImage object
+
+        Convert input to a binary image by setting all data with signal-to-noise
+        below some threshold to zero, and all data above the threshold to one.
+        If the imageBinning config parameter has been set, this procedure will
+        be preceded by a weighted binning of the data.
+
+        Parameters
+        ----------
+        maskedImage : `lsst.afw.image.Exposure`
+            Image to be (optionally) binned and converted
+
+        Returns
+        -------
+        out_data : `np.ndarray`
+            2-d binary image of pixels above the signal-to-noise threshold.
+        """
+        binning = self.config.imageBinning
+
+        data = maskedImage.image.array
+        weights = 1 / maskedImage.variance.array
+        mask = maskedImage.getMask()
+
+        detectionMask = ((mask.array & mask.getPlaneBitMask("DETECTED")))
+        badMask = ((mask.array & mask.getPlaneBitMask("NO_DATA")) |
+                   (mask.array & mask.getPlaneBitMask("INTRP")) |
+                   (mask.array & mask.getPlaneBitMask("BAD")) |
+                   (mask.array & mask.getPlaneBitMask("SAT")) |
+                   (mask.array & mask.getPlaneBitMask("EDGE"))).astype(bool)
+
+        fitMask = detectionMask.astype(bool) & ~(badMask.astype(bool))
+        fitData = np.copy(data)
+        fitData[~fitMask] = 0
+        fitWeights = np.copy(weights)
+        fitWeights[~fitMask] = 0
+
+        if binning is not None:
```

Comment:
In the long run, this isn't the right place for a binning algorithm.  Consider opening a ticket to move this lower into the stack.  In the stack we have `afwMath.binImage()` which can only handle means.   Most the image binning used by the stack is actually in`afwMath.makeBackground` which can do clipped means and medians.  https://github.com/lsst/afw/blob/master/doc/lsst.afw.math/Background-example.rst

Also, why do you need inverse-variance weighted means when binning this image?

Reply:
I looked at using `afwMath.binImage()` here, but I think it would be more convoluted. Even without doing a weighted mean, you need the binned weights in order to get the S/N at the end. I think this means that, to use `afwMath.binImage()`, you would have make an afwImage `ImageF` or similar, put the weights in that, and run `afwMath.binImage()` on that. 

Separately, doing a weighted mean takes care of removing bad pixels, though it is probably not crucial otherwise.